src/AutoRest.CSharp/Output/Models/RestClientBuilder.cs

```diff
@@ -21,150 +21,47 @@
 
 namespace AutoRest.CSharp.Output.Models
 {
-    internal class RestClient : ClientBase
+    internal class RestClientBuilder<T> where T: OutputLibrary
     {
-        private static readonly HashSet<string> IgnoredRequestHeader = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
-        {
-            "x-ms-client-request-id",
-            "tracestate",
-            "traceparent"
-        };
-
-        protected string RestClientSuffix { get; }
-
-        private readonly OperationGroup _operationGroup;
-        private readonly BuildContext _context;
-        private readonly Dictionary<string, Parameter> _parameters;
         private readonly SerializationBuilder _serializationBuilder;
-        private Dictionary<ServiceRequest, RestClientMethod>? _requestMethods;
-        private Dictionary<ServiceRequest, RestClientMethod>? _nextPageMethods;
-        private RestClientMethod[]? _allMethods;
+        private readonly BuildContext<T> _context;
+        private readonly Dictionary<string, Parameter> _parameters;
+        private readonly Func<RequestParameter, bool> _methodFilter;
 
-        public RestClient(OperationGroup operationGroup, BuildContext context) : base(context)
+        public RestClientBuilder (OperationGroup operationGroup, BuildContext<T> context, Func<RequestParameter, bool>? methodFilter)
         {
-            _operationGroup = operationGroup;
+            _serializationBuilder = new SerializationBuilder ();
             _context = context;
+            _methodFilter = methodFilter ?? new Func<RequestParameter, bool>(p => true);
```

Comment:
Sounds more like a parameter filer to me, is it worth taking it into the BuildMethod call or can we pre-filter the collection we pass into the BuildMethod?